server/boardComponent/entities/forum/api.js

```diff
@@ -11,34 +11,36 @@ const getDiscussions = require('./controller').getDiscussions
  */
 const router = express.Router() // eslint-disable-line
 
-  // get all forums
+// get all forums
 router.route('/').get(
   expressJwt({ secret: config.jwtSecret, requestProperty: 'auth' }),
   (req, res) => {
-  getAllForums().then(
+    getAllForums().then(
       (result) => { res.send(result) },
       (error) => { res.send(error) }
     )
-})
+  })
 
-  // get discussions of a forum
+// get discussions of a forum
 router.route('/:forum_slug/discussions').get(
   // expressJwt({ secret: config.jwtSecret, requestProperty: 'auth' }),
   (req, res) => {
-  getDiscussions(req.params.forum_slug, false, req.query.sorting_method, req.query.q).then(
+    const { q, offset, limit } = req.query
+    getDiscussions(req.params.forum_slug, false, req.query.sorting_method, q, offset, limit).then(
       (result) => { res.send(result) },
       (error) => { res.send([]) }
     )
-})
+  })
 
-  // get pinned discussions of a forum
+// get pinned discussions of a forum
 router.route('/:forum_slug/pinned_discussions').get(
   // expressJwt({ secret: config.jwtSecret, requestProperty: 'auth' }),
   (req, res) => {
-  getDiscussions(req.params.forum_slug, true).then(
+    const { q, offset, limit } = req.query
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the declaration of the unused 'q' variable. ([javascript:UnusedVariable](https://sonarcloud.io/organizations/aumanjoa-github/rules#rule_key=javascript%3AUnusedVariable))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=chronas-api&pullRequest=29&issues=AWbY54uASgN4uCclP4pi&open=AWbY54uASgN4uCclP4pi)

[ISSUE_KEY]: AWbY54uASgN4uCclP4pi